### PR TITLE
[7X] Fix psycopg2 encoding issue

### PR DESCRIPTION
**Pull request by higuoxing**
psycopg2's `getquoted()` API returns `latin-1` encoded binary string by default which is causing unexpected failures to some gpMgmt tools including gpload, analyzedb, minirepro. This patch helps fix it by teaching psycopg2's `QuotedString` adapter use `utf-8` encoding.

Reproducing steps for `analyzedb`:

Create a table with special name.

```sql
postgres=# create table spiegelungssätze(i int);
```

Run `analyzedb` against the postgres db.

```bash
$ analyzedb -d postgres
```

Backtrace:

```
➜ analyzedb -d postgres
20230910:21:51:11:689552 analyzedb:laptop:v-[INFO]:-Starting analyzedb with args: -d postgres
20230910:21:51:11:689552 analyzedb:laptop:v-[INFO]:-Getting and verifying input tables...
20230910:21:51:11:689552 analyzedb:laptop:v-[INFO]:-Checking for tables with stale stats...
20230910:21:51:11:689552 analyzedb:laptop:v-[ERROR]:-'utf-8' codec can't decode byte 0xe4 in position 21: invalid continuation byte
Traceback (most recent call last):
  File "/home/v/.local/gpdb7/bin/analyzedb", line 376, in execute
    heap_partitions = get_heap_tables_set(self.conn, input_tables_set)  # set((schema1,table1), ...])
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/v/.local/gpdb7/bin/analyzedb", line 1004, in get_heap_tables_set
    oid_str = get_oid_str(input_tables_set)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/v/.local/gpdb7/bin/analyzedb", line 976, in get_oid_str
    return ','.join(map((lambda x: regclass_schema_tbl(x[0], x[1])), table_list))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/v/.local/gpdb7/bin/analyzedb", line 976, in <lambda>
    return ','.join(map((lambda x: regclass_schema_tbl(x[0], x[1])), table_list))
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/v/.local/gpdb7/bin/analyzedb", line 984, in regclass_schema_tbl
    return "to_regclass('%s')" % (escape_string(schema_tbl))
                                  ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/v/.local/gpdb7/lib/python/gppylib/utils.py", line 515, in escape_string
    return adapted.getquoted().decode()[1:-1]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xe4 in position 21: invalid continuation byte
20230910:21:51:11:689552 analyzedb:laptop:v-[CRITICAL]:-analyzedb failed. (Reason=''utf-8' codec can't decode byte 0xe4 in position 21: invalid continuation byte') exiting...
```

This patch depends on #16406.

### CODE REVIEW

File: gpMgmt/bin/gpload_test/gpload2/TEST_local_schema_and_mode.py [hunk 1]
```diff
@@ -907,3 +907,13 @@ def test_547_gpload_insert_staging_without_DK():
     f = open(TestBase.mkpath('query547.sql'), 'a')
     f.write("\\! psql -d reuse_gptest -c '\\d staging_gpload_reusable_*'")
     f.close()
+
+@TestBase.prepare_before_test(num=548, times=1)
+def test_548_gpload_ext_staging_table_with_externalschema():
+    "548 gpload reuse ext_staging_table if it is configured with externalschema"
+    setup_file = TestBase.mkpath('setup.sql')
+    TestBase.runfile(setup_file)
+    with open(TestBase.mkpath('query26.sql'),'a') as f:
```
Comment (xiaoxiaoHe-E) [inline gpMgmt/bin/gpload_test/gpload2/TEST_local_schema_and_mode.py:916]: Are you trying to add this query sql to file query548.sql?

Comment (higuoxing) [reply gpMgmt/bin/gpload_test/gpload2/TEST_local_schema_and_mode.py:916]: Ah, right. I want to add it to query548.sql.

**Review by xiaoxiaoHe-E — APPROVED** [review]

**Review by piyushc01 — APPROVED** [review]
Looks good from the cluster management point of view. 
Make sure you have run the pipeline on supported platforms.

File: gpMgmt/bin/gpload.py [hunk 2]
```diff
@@ -548,8 +548,16 @@ def is_keyword(tab):
...
+# Escape single quotes, backslashes appearing in the string according to the SQL string constants syntax.
+# E.g.,
+# >>> escape_string(r"O'Reilly")
+# "O''Reilly"
 def escape_string(string):
-    return psycopg2.extensions.QuotedString(string).getquoted()[1:-1].decode()
+    adapted = psycopg2.extensions.QuotedString(string)
+    # The getquoted() API returns 'latin-1' encoded binary string by default, we need to specify
+    # the encoding manually.
+    adapted.encoding = 'utf-8'
```
Comment (kmacoskey) [inline gpMgmt/bin/gpload.py:559]: @higuoxing  Is this potentially an issue? Does this hardcode gpload assuming `utf-08` locale?

Comment (higuoxing) [reply gpMgmt/bin/gpload.py:559]: > Is this potentially an issue?

I think it's not a potential issue.

> Does this hardcode gpload assuming utf-08 locale?

No, it just indicates that `adapted.getquote()` returns a utf-8 encoded binary string. Eventually, this function returns a valid Python `str` object, Python itself knows how to deal with encodings. It doesn't affects or be affected by either the server encoding or the clients encoding.

**Pull request merged**